clients/src/main/java/org/apache/kafka/clients/producer/internals/RecordBatch.java

```diff
@@ -67,10 +68,13 @@ public FutureRecordMetadata tryAppend(long timestamp, byte[] key, byte[] value,
         if (!this.records.hasRoomFor(key, value)) {
             return null;
         } else {
-            this.records.append(offsetCounter++, timestamp, key, value);
+            long checksum = Record.computeChecksum(timestamp, key, value, CompressionType.NONE, 0, -1);
```

Comment:
This is a checksum for inner record, which is not compressed. If you follow the original path of MemoryRecords.append, that's how checksum is computed. I just moved that same computation couple of levels up.